### ISSUE TITLE
Training Costs for MapAnything

### ISSUE THREAD

**Issue by Nik-V9** [HengyiWang/amb3r#5]
Hey @HengyiWang , congrats on the great work! Excited to try out the model soon. I wanted to flag a misunderstanding with the training costs of MapAnything, particularly in comparison to VGGT.

The AMB3R paper states that 1 H100 hour is equivalent to 3 A100 hours. Note that this would be only true if the codebase uses Flash Attention (FA) 3 over FA 2. For MapAnything, we use torch sdpa which defaults to FA 2.

Furthermore, given the same amount of training time (10 days), while we use a H200 (140 GB) vs H100 (80 GB), MapAnything gradient descent steps is 420K vs VGGT gradient descent steps is 160K. This is enabled by our 2x smaller model, more efficient codebase and use of **no gradient checkpointing for the multi-view transformer** (MapAnything - 24 ViT-L Encoder Layers + 24 (in total) ViT-B AA Transformer Layers Vs. VGGT - 24 ViT-L Encoder Layers + 48 (in total) ViT-L AA Transformer Layers).

When trained on A100 nodes, we find the training time for MapAnything to be the same, i.e., 10 days of 420K steps (2x smaller batch size than the H200) with very marginal decrease in convergence/performance.

Overall, tldr, MapAnything is the same training cost as VGGT (this is something the VGGT authors also agree with) and with out explicit mention of FA3, I would presume the H100s to be equivalent to the 1 -1.5 A100s - maybe being faster by a factor of up to 1.5x based on multi-node NVLink infra. I would appreciate it if you can update your training cost plots on Arxiv and during your full public release/announcement.

**Comment by HengyiWang** [comment]
Hi @Nik-V9, sorry for the late reply. Thank you for the detailed breakdown of the training costs. This is really helpful!


Our original estimation of 1H100/H200 hour (amb3r/mapany) equals to 3 A100 hours (dust3r/vggt/pi3) was based on this [NVIDIA blog](https://developer.nvidia.com/blog/nvidia-hopper-architecture-in-depth/). However, I fully agree that the real-world performance often varies due to system-level overheads, packages, etc.

For the final version, we are happy to revise the training cost comparison as follows:

1) Adjust the conversion rate to ~2 A100 hours
2) Differentiate columns of models trained on A100/H100/H200 with distinct patterns/shades
3) Add a footnote pointing to this GitHub issue so that readers can see the full context

Feel free to reach out if you have further suggestions or spot anything else that could be clarified:)
